tools/c7n_gcp/c7n_gcp/resources/kms.py

```diff
@@ -187,3 +189,53 @@ def get(client, resource_info):
         @classmethod
         def _get_location(cls, resource):
             return resource["name"].split('/')[3]
+
+        @classmethod
+        def _get_location(cls, resource):
+            return resource["name"].split('/')[3]
+
+
+@resources.register('kms-location')
+class KmsLocation(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'cloudkms'
+        version = 'v1'
+        component = 'projects.locations'
+        enum_spec = ('list', 'locations[]', None)
+        scope = None
+        name = id = 'name'
+        default_report_fields = [
+            "name", "createTime"]
+        asset_type = "cloudkms.googleapis.com/Location"
+
+    def _fetch_resources(self, query):
+        super_fetch_resources = QueryResourceManager._fetch_resources
+        session = local_session(self.session_factory)
+        project = session.get_default_project()
+        return super_fetch_resources(self, {'name': 'projects/{}'.format(project)})
+
+
+@KmsLocation.filter_registry.register('kms-keyring')
```

Comment:
given its already being used on a kms-location, the kms prefix on the filter feels redundant.

Reply:
done